refinery/tool_manager/models.py

```diff
@@ -237,38 +243,88 @@ def __str__(self):
     def launch(self):
         if self.get_tool_type() == ToolDefinition.VISUALIZATION:
             container = DockerContainerSpec(
-                image_name=self.tool_definition.docker_image_name,
+                image_name=self.tool_definition.image_name,
                 container_name=self.container_name,
                 labels={self.uuid: ToolDefinition.VISUALIZATION},
                 container_input_path=(
                     self.tool_definition.container_input_path
                 ),
-                input={"file_relationships": self.file_relationships}
+                input={"file_relationships": eval(self.file_relationships)},
+                manager=self.get_django_docker_engine_manager()
             )
             try:
                 container.run()
             except APIError as e:
                 return HttpResponseServerError(content=e)
             else:
-                return HttpResponse(self.get_relative_container_url())
+                return JsonResponse(
+                    {
+                        "tool_url": self.get_relative_container_url()
+                    }
+                )
 
         if self.get_tool_type() == ToolDefinition.WORKFLOW:
             raise NotImplementedError
 
+    def get_django_docker_engine_manager(self):
```

Comment:
I don't think it belongs in a class because it doesn't need to operate on any of the class properties.